cmd/nfd-master/main.go

```diff
@@ -109,6 +112,10 @@ func argsParse(argv []string) (master.Args, error) {
 
 	// Parse argument values as usable types.
 	var err error
+	args.Instance = arguments["--instance"].(string)
+	if ok, _ := regexp.MatchString(`^([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]$`, args.Instance); args.Instance != "" && !ok {
+		return args, fmt.Errorf("invalid --instance %q", args.Instance)
```

Comment:
It would be good to add some help message here or in line 80 and provide details what instance name format is expected

Reply:
Indeed 😄  Added an explanation to the error message